cmd/manager/main.go

```diff
@@ -107,6 +109,21 @@ func main() {
 	log.Print("Starting the Cmd.")
 	stopCh := signals.SetupSignalHandler()
 
+	// set ClusterOperator status to report Migration
+	status.ReportMigration(cfg, mgr, namespace, os.Getenv("RELEASE_VERSION"), stopCh)
+
+	client, err := client.New(cfg, client.Options{})
+	if err != nil {
+		exit(err)
+	}
+
+	// Perform migration logic to upgrade cluster from 4.1.z to 4.2.z
+	migrator := migrator.NewMigrator(client)
+	err = migrator.Migrate(namespace)
+	for err != nil {
```

Comment:
I was thinking we would attempt, repeatedly, in the foreground, until we either give up or succeed.

This is what we're doing for a similar migration in olm: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/cmd/olm/cleanup.go#L23

The difference here is that marketplace can technically keep working if this migration doesn't succeed. But I would think we should still block here because we know that if this migration doesn't run, there will be a problem for a future update (unless we intend to keep CSCs around forever?) Correct me if I'm wrong.

Reply:
I agree with this approach. I just want to clarify that we do not delete all CSCs as part of this migration - only the CSCs created by the operator or by the UI. In the case of user-created CSCs, the migration code will just ignore their existence (note the operator will still be able to act on these properly). Although we are able to delete them if we want to, the argument was made that it was a bad user experience to have resources created in 4.1 disappear in 4.2. Anyway if the migration code doesn't run I believe we can still manage to delete all CSCs as part of a 4.2 to 4.3 migration if we decide to depricate CSCc in 4.3+.